### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by yiahmed**
# Submitting Week 7 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ✅ ] Coding Challenges
- [ ✅ ] Coding Concepts
- [ ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -22,25 +37,97 @@ const people = [
...
+// Add the word is in the string interpolation followed by values.occupation to add the last part of string, finally add a period and close the string interpolation.
+
+// Raw Code following the pseudo
+const sentenceCap = (array) => {
+   return Object.values(array).map(value => {
+  return `${value.name.split(" ").map(value => value[0].toUpperCase() + value.substring(1)).join(" ")} is ${value.occupation}.`
+  })
+}
+
+// Refactored code using deconstruction to access name and occupation directly
```
Comment (elysemontano) [inline code-challenges.test.js:53]: 🎉 Excellent work!  Love this refactor and detailed pseudocode!

File: interview-questions.md [hunk 2]
```diff
@@ -8,36 +8,76 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Your answer: The names of some of the rails generator commands I use pretty often are: rails g migration, rails g controller, and rails g model.
 
-Researched answer:
+rails g migration: Generates a migration within the migrate directory that contains the migration name and attributes. Migrations are used to update/change the database in different ways. To execute the changes when you are done in the migrate directory you would run rails db:migrate in the console.
+
+rails g controller: Running this command in the terminal creates a new controller within the controllers directory. Optional Action methods can be added to the controller file.
+
+rails g model: Running this command create a model within the models directory and also generates a migration file in the migrate directory to create a data base. Attributes can be specified when running this command as well to create columns within the database. You can find the structure of the database within the schema.
+
+Researched answer: The name of three commonly used rails generator commands are: rails g controller, rails g migrate, rails g model:
```
Comment (elysemontano) [inline interview-questions.md:31]: Excellent work on this question, I would use the researched answer to look into some other generate commands and talk about what they do such as resource and scaffold are pretty cool to talk about.

**Review by elysemontano — APPROVED** [review]
Outstanding work in this assessment!!  1000% crushed this!  One thing to keep an eye on is indentation, but your answers and logic are top notch.  If you want to challenge yourself a little more in the code challenges, try thinking of some edge cases and then writing tests as well.
